flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/DefaultSchedulerTest.java

```diff
@@ -1618,6 +1621,141 @@ public void testCheckpointCleanerIsClosedAfterCheckpointServices() throws Except
         }
     }
 
+    @Test
+    public void testJobStatusHookWithJobFailed() throws Exception {
```

Comment:
The three test methods seem quite similar, could we extract a common help method?